rfd/0213-relay-a-lightweight-proxy.md

```diff
@@ -0,0 +1,64 @@
+
+The configuration also includes an agent connection count number that works the same as the agent connection count for proxy peering (i.e., with an agent connection count of 2 it's expected that there will always be at least 2 relays in the group and agents will maintain reverse tunnels to 2 different relays in the group). The count will also be included in the resource heartbeat for debugging purposes.
+
+Relays will route connections from clients to agents within the same subenvironment, as well as client connections from the broader Teleport cluster (from a proxy through the relay's own reverse tunnels). Client connections from the subenvironment to the relays are never going to be forwarded to the broader cluster if the agent that the client is trying to reach is not connected to the relay group. All Teleport API access (agent joining, heartbeats, audit logging and session recording upload, client login), as well as use of the Teleport web UI, will use the regular Teleport control plane directly, and will not be handled or forwarded by the relays - this is mostly to keep the required permissions for the `Relay` role as narrow as possible, but it will also simplify the implementation and it will avoid adding another potential point of failure for internal connectivity.
+
+This RFD will detail support for (agentful) SSH and Kubernetes, other protocols are out of scope - what _can_ be supported in the future varies based on the protocol and the class of client used to access it: everything should be supportable through a local `tsh proxy` (and/or Teleport Connect and/or Teleport VNet), direct client connections for Database access might be possible, App access seems harder to support (as it's heavily tied to the web UI, in its direct client connection mode).
+
+### Agent configuration and behavior
+
+Agents will be configured with a `proxy_server` address pointed at the usual Proxy public address (which is still going to be used for the initial joining process and for all Teleport API access), but they can optionally also include a `relay_server` pointing at the relay group address. If such an address is specified, agents will not open connections for the regular reverse tunnel.
```

Comment:
Should we also have an (optional?) `relay_group` setting to reject connecting to a relay if the group doesn't match what the agent expects?